set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -642,86 +596,45 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
-			// check client access (a valid "wskey" must be provided)
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
-
-			// authorize user or if unauthorized respond with HTTP 403
-			getAuthorizationService().authorizeUser(userToken, wsKey, identifier, Operations.DELETE);
+		    // verify access rights
+		    verifyWriteAccess(Operations.DELETE, request);
 
 			// retrieve a user set based on its identifier
 			// if the Set doesn’t exist, respond with HTTP 404
 			UserSet existingUserSet = getUserSetService().getUserSetById(identifier, false);
 
-			// check that only the admins and the owners of the user sets are allowed to delete the user set. 
-			// in the case of regular users (not admins), the autorization method must check if the users 
-			// that calls the deletion (i.e. identified by provided user token) is the same user as the creator 
-			// of the user set
-			hasModifyRights(existingUserSet, wsKey, userToken);
-
 			// check timestamp if provided within the "If-Match" HTTP header, if false respond with HTTP 412
-			checkHeaderTimestamp(request, existingUserSet);
-						
+		    checkIfMatchHeader(existingUserSet.getModified().hashCode(), request);
+
 			// if the user set is disabled and the user is not an admin, respond with HTTP 410
 			HttpStatus httpStatus = null;
-			if (existingUserSet.isDisabled()) {
-				if (!isAdmin(wsKey, userToken)) { 
-					// if the user is the owner, the response should be 410
-					if (isOwner(existingUserSet, userToken)) {
-//						httpStatus = HttpStatus.GONE;	
-						throw new OperationAuthorizationException(I18nConstants.USERSET_ALREADY_DISABLED, 
-								I18nConstants.USERSET_ALREADY_DISABLED, 
-								new String[]{existingUserSet.getIdentifier()},
-								HttpStatus.GONE);
-						
-					} else {
-						// if the user is a registered user but not the owner, the response should be 401 (unathorized)
-						httpStatus = HttpStatus.UNAUTHORIZED;					
-					}
-				} else {
-					// if the user is admin, the set should be permanently deleted and 204 should be returned
-					getUserSetService().deleteUserSet(existingUserSet.getIdentifier());
-					httpStatus = HttpStatus.NO_CONTENT;
-				}
-			} else {			
-				// if the user is an Administrator then permanently remove item 
-				// (and all items that are members of the user set)
-				 httpStatus = HttpStatus.NO_CONTENT;
-				 if (isAdmin(wsKey, userToken)) {
-					 getUserSetService().deleteUserSet(existingUserSet.getIdentifier());
-				 } else { // otherwise flag it as disabled
-  					 if (isOwner(existingUserSet, userToken)) {
-  						 getUserSetService().disableUserSet(existingUserSet);
-  					 } else {
- 						// if the user is a registered user but not the owner, the response should be 401 (unathorized)
- 						httpStatus = HttpStatus.UNAUTHORIZED;					
-  					 }
-				 }
-			}			
+
+		    if (existingUserSet.isDisabled()) {
```

Comment:
please confirm if this behaviour was added to specs

Reply:
behaviour remains the same